pkg/util/provider/machinecontroller/machine_bootstrap_token.go

```diff
@@ -90,11 +91,15 @@ func (c *controller) getBootstrapTokenOrCreateIfNotExist(ctx context.Context, ma
 			if err != nil {
 				return nil, err
 			}
+			expiration := c.safetyOptions.MachineCreationTimeout.Duration
+			if machine.Spec.MachineConfiguration != nil && machine.Spec.MachineConfiguration.MachineCreationTimeout != nil {
+				expiration = machine.Spec.MachineConfiguration.MachineCreationTimeout.Duration
+			}
```

Comment:
we can use the helper function `getEffectiveCreationTimeout` for this.

Reply:
ah didn't know that helper function exists.

Updated the code PTAL.